### ISSUE TITLE
How these docs should be done.

### ISSUE THREAD

**Issue by Registrierungsgenervter** [TYPO3-Documentation/DocsTypo3Org-Homepage#64]
Here is my opinion as a newbie who just has resigned learning this framework:

* merge [all these documents](https://docs.typo3.org/) into one, shorten it heavily. Makes it easier for the readers and easier for devs to update them. A good idea is to have automated tests for the code examples.
* Create a full clone of this document for every new framework version, update the docs _before_ you release the new version.

See the Python docs as a reference.

**Comment by Registrierungsgenervter** [comment]
[Related](https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/430).

**Comment by sypets** [comment]
Thank you for taking the time to give feedback. I think you made some good points but obviously, we cannot give you an answer right now or change things overnight.

**Comment by Registrierungsgenervter** [comment]
Yes, of course, this is a big change and probably the 'Chief' would have to write v1.

**Comment by sypets** [comment]
I would like to follow up on your suggestions some more.

First, however, I would like to point out: TYPO3 documentation is mostly written and maintained by voluntary contributors. Everyone in the community is welcome and urged to participate. The documentation is apparently regarded as important by the community, but not so many are willing to contribute. We are doing the best we can right now, with the resources we have.  

Also: Feedback is really, really important and I don't want to blow you off. On the contrary, I would like to invite you to stay a little longer and participate. And: You can also talk about documentation issues on [Slack](https://docs.typo3.org/typo3cms/HowToDocument/HowToGetHelp.html)

I realize TYPO3 is not a simple thing to learn, but one of the reasons it is so difficult are also a big fat PLUS: It gives you a lot of freedom, there is not just one way to do things, you can configure the hell out of it, you can add your own extensions, you can [participate in core development](https://typo3.org/community/teams/typo3-development/), etc. etc. See other [features](https://typo3.org/cms/features/).

**About your suggestions:**

> merge all these documents into one, shorten it heavily. 

That is something that has already been happening. 

* The previous guides "Inside TYPO3", "FAL", "Services" and "Security Guide" have all been merged into "TYPO3 Explained" by Christian Kuhn, see this [news](https://docs.typo3.org/News/2018/Index.html#typo3-explained-and-more) for more information.
* On a smaller scale: If we see duplicate information (which is often the case, unfortunately), we merge it into one place and add links
* The same goes for the Wiki. There is a https://wiki.typo3.org that is partly outdated. There are a number of pages that have already been merged. And there are more in my [pipeline](https://github.com/orgs/TYPO3-Documentation/projects/3).

However, merging it **all** into only one repository (or core) is not an approach I would necessarily take right now. Having a set of smaller pieces (e.g. individual tutorials) has its merits. That is a discussion that is beyond the scope of one issue, though. 

I do agree, that the goal should be to deduplicate, shorten some parts, completely remove unhelpful or outdated stuff and try to make it more homogenous and user-friendly. 

> Makes it easier for the readers and easier for devs to update them. 

How? Right now, all the reader needs is a GitHub account and he / she can press the "Edit me on GitHub" button. We have recently updated our documentation, there is even a video posted on TYPO3 YouTube Channel, sponsored by TYPO3 GmbH, see https://docs.typo3.org/typo3cms/HowToDocument/WritingDocsOfficial/Index.html.

Additionally, we have a Docker image for rendering locally. So if you want to make bigger changes, you can do it in your IDE. Not many open source projects have that. 

Ok, there are some things: GitHub preview does not support all the Sphinx syntax (only reST), having a proper wysiwyg editor might make it even easier, etc. etc. 

If you use individual Markdown files, you get better rendering on GitHub and possibly better tool support, but Markdown has some other problems. And you cannot, for example, automatically create a toc (table of contents) and full blown menu. 

As for the devs:
* About the **Changelog**: breaking changes, features, deprecations etc. must have a changelog entry. The changelog is actually maintained in the core repository. When you write a patch with a breaking change, deprecation or new feature you are obligated to add a .rst file as Changelog. There is even a  tool for creating that: https://forger.typo3.com/utility/rst (thanks to Mathias Schreiber, TYPO3 GmbH). Here too, the docs can be rendered locally. The changelog is then [automatically rendered](https://docs.typo3.org/typo3cms/extensions/core/latest/), it can be viewed in the Install Tool and the [Extension Scanner.](https://docs.typo3.org/typo3cms/CoreApiReference/ApiOverview/ExtensionScanner/Index.html). I think that is pretty brilliant (thanks again to Christian Kuhn and others) and I don't know any other open source project who does it like that. Additionally, every single Changelog entry can be edited on GitHub as well (just press "Edit me on GitHub"). Can it be made easier? Can the sky be even bluer? Certainly, but time is not an infinite resource. 

> A good idea is to have automated tests for the code examples.

Good point!  I created a new issue to follow up on this: https://github.com/TYPO3-Documentation/T3DocTeam/issues/69

> Create a full clone of this document for every new framework version, update the docs before you release the new version.

Absolutely agree. We should not put documentation online that is "unfinished". Releasing it with the next release also makes sense.

I also created an issue for this because I think it is important: https://github.com/TYPO3-Documentation/T3DocTeam/issues/70

> See the Python docs as a reference.

I will do that. On the other end of the pole: See Linux contribution and git man pages.

**Comment by susannemoog** [comment]
I'm a bit overwhelmed by this issue - how can we continue here? For the original request: the docs are now kept up-to-date and every new release has (mostly) updated documentation. The restructuring is a work in progress and cannot be solved with a single task. What concrete todo do we want to tackle with this issue or would it be ok to create / link to additional concrete issues and close this one?
